test/e2e/apps/job.go

```diff
@@ -290,18 +290,8 @@ var _ = SIGDescribe("Job", func() {
 		framework.ExpectNoError(err, "failed to confirm that pods aren't created for job %s in namespace %s", job.Name, f.Namespace.Name)
 
 		ginkgo.By("Checking Job status to observe Suspended state")
-		job, err = e2ejob.GetJob(ctx, f.ClientSet, f.Namespace.Name, job.Name)
-		framework.ExpectNoError(err, "failed to retrieve latest job object")
-		exists := false
-		for _, c := range job.Status.Conditions {
-			if c.Type == batchv1.JobSuspended {
-				exists = true
-				break
-			}
-		}
-		if !exists {
-			framework.Failf("Job was expected to be completed or failed")
-		}
+		err = e2ejob.WaitForJobSuspend(ctx, f.ClientSet, f.Namespace.Name, job.Name)
```

Comment:
this should probably return the last version of the job to be used below.

Reply:
yea, good call. Fixed.

Reply:
I had to revert that change.  We got errors saying that the job was not the latest and that caused a failure downstream in the job.  I removed the return and I retrieve the job before patching it to unsuspend.